spec/jobs/deposit_job_spec.rb

```diff
@@ -47,6 +47,18 @@
       expect(SdrClient::Deposit::UploadFiles).to have_received(:upload)
       expect(DepositStatusJob).to have_received(:perform_later).with(object: work_version, job_id: 1234)
     end
+
+    context 'when the deposit is for a PURL reservation' do
+      let(:work_version) do
+        build(:work_version, :reserving_purl, work: work)
+      end
+
+      # TODO: failing because purl reservation placeholder type is not valid cocina type
```

Comment:
not sure what the right thing to do here is.  add a placeholder type to the allowable cocina types?  allow null work type in H2?  something else i'm not thinking of?

also no opinion as to whether this PR should be blocked over this shortcoming.  probably, but i could also see some value in getting it merged since it's been open for a while.  it's not huge, but it's not small, and it's been through a couple significant rebases.

Reply:
fixed by https://github.com/sul-dlss/happy-heron/pull/1223